doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,118 @@
+```
+
+Note that the second option could be not supported for some Azure SDK clients if it doesn't make sense for the service behavior (like Key Vault client).
+
+The Azure SDK for C++ client `will hold an internal copy of the options` passed in the arguments. This means that the options object used to init the client can be disposed. However, since the HTTP transport adapter is a shared ptr, it won't be disposed until any client using it is disposed first.
+
+## HTTP Transport Adapter Options
+The libcurl and WinHTTP transport adapters can also be init with specif options that control features from the HTTP stack underneath it.
+For example, you can set a PROXY or override the default CA path for libcurl. See next example.
+```cpp
+  /*
+  * Customize libcurl http transport adapter.
+  */
+
+  // Create HTTP transport adapter options and set options.
+  Azure::Core::Http::CurlTransportOptions curlTransportOption;
+  curlTransportOption.Proxy = "https://my.proxy.com";
+  // Create HTTP transport adapter with options.
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlTransportOption);
+
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = curlTransportAdapter; 
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+The HTTP transport adapter are specific for each adapter and are optional. This means that if you are creating your own HTTP transport adapter, you don't need to create options for it if you don't want to.
+
+
+## Building a custom HTTP Transport Adapter
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required to be set when building your own HTTP transport adapter.
+
+Follow next steps to implement your own HTTP Transport Adapter.
+
+1. Implement interface. The HTTP transport adapter interface is define in `Azure::Core::HTTP::HttpTransport` from `inc/azure/core/http/transport.hpp`.
+
+```cpp
+/*
+* Implementing the HTTP transport adapter.
+*/
+
+#include "azure/core/http/transport.hpp"
+
+// Derive from HttpTransport
+class CurlTransport : public Azure::Core::Http::HttpTransport {
+  public:
+    // Override the Send method
+    std::unique_ptr<Azure::Core::Http::RawResponse> Send(
+        Azure::Core::Context const& context,
+        Azure::Core::Http::Request& request) override;
+};
+```
+
+The implementation needs to take care of using the data from the argument `request` containing all the HTTP request information to write the HTTP request using some C++ HTTP stack. Then it needs to create the raw response from the response and return it. The `context` argument can be optionally used to check if the request is cancelled before performing I/O operations (like writing to the network). 
```

Comment:
We may want to call out the need to return the response as streaming data and wrapping that in a bodystream and provide some example.

Reply:
Yes, I though about it but I considered that we need to create a new document for how the bodyStream and the responses work and then link to this section.  But yes, you are right. that needs to be said.